test/mini-graphql-http-client.js

```diff
@@ -93,4 +93,116 @@ describe("MiniGraphqlHttpClient tests", () => {
             expect(cache2.map.get("2421565178")).to.have.deep.property("json", { bla: 1 });
         });
     });
+
+    const createBadAndGoodFakeFetch = () => {
+        function fakeFetch() {
+            const responseArray = [
+                { ok: false, status: 500, json: () => ({ bla: 1 }) },
+                { ok: false, status: 500, json: () => ({ bla: 1 }) },
+                { ok: false, status: 500, json: () => ({ bla: 1 }) },
+                { ok: false, status: 500, json: () => ({ bla: 1 }) },
+                { ok: true, status: 200, json: () => ({ bla: 1 }) },
+                { ok: false, status: 400, json: () => ({ bla: 1 }) },
+                { ok: true, status: 200, json: () => ({ bla: 1 }) },
+            ];
+            fakeFetch.calls += 1;
+            return responseArray[fakeFetch.calls];
+        }
+        fakeFetch.calls = -1;
+        return fakeFetch;
+    };
+
+    describe("#Retry", () => {
+        it("should make 6 retries on 5xx error response, but stops on 5 try, because it's get good response", async function () {
+            let wasError = false;
+            let numberOfRetries = 6;
+            const cache = {};
+            const fakeFetch = createBadAndGoodFakeFetch();
+            const client = MiniGraphqlHttpClient({
+                uri: "https://a.aa",
+                fetch: fakeFetch,
+                cache,
+                retry: numberOfRetries,
+            });
+
+            await client.query({ query: "{ bla }" }).catch(() => {
+                wasError = true;
+            });
+
+            expect(fakeFetch.calls).to.be.equal(4);
+            expect(wasError).to.be.false;
+        });
+
+        it("should make 3 retries on 5xx error response and return error", async function () {
+            let wasError = false;
+            let numberOfRetries = 3;
+            const cache = {};
+            const fakeFetch = createBadAndGoodFakeFetch();
+            const client = MiniGraphqlHttpClient({
+                uri: "https://a.aa",
+                fetch: fakeFetch,
+                cache,
+                retry: numberOfRetries,
+            });
+
+            await client.query({ query: "{ bla }" }).catch(() => {
+                wasError = true;
+            });
+
+            expect(fakeFetch.calls).to.be.equal(3);
+            expect(wasError).to.be.true;
+        });
+
+        it("should make 3 retries on 4xx error response and return error after 1st try", async function () {
```

Comment:
Bad test name. "should make reties on 4xx response".
The code must **not** make any retries if the error is 4xx.